core/node/api_server/src/web3/namespaces/eth.rs

```diff
@@ -404,14 +404,22 @@ impl EthNamespace {
         // Check if the bytecode is an EVM bytecode, and if so, pre-process it correspondingly.
         let marker = BytecodeMarker::new(contract_code.bytecode_hash);
         let prepared_bytecode = if marker == Some(BytecodeMarker::Evm) {
-            trim_padded_evm_bytecode(&contract_code.bytecode)
-                .with_context(|| {
-                    format!(
-                        "malformed EVM bytecode at address {address:?}, hash = {:?}",
+            trim_padded_evm_bytecode(
+                BytecodeHash::try_from(contract_code.bytecode_hash).unwrap_or_else(|_| {
+                    panic!(
```

Comment:
Nit: I'd return an error here similarly to the call below; not clear why this place merits a panic, but `trim_padded_evm_bytecode()` doesn't.